### PR TITLE
feat: add IsUrl

### PR DESCRIPTION
**Pull request by osintalex**
Here's a first go at this. I didn't write too many test cases since most of the Url checking logic is already tested in Pydantic.

### CODE REVIEW

File: dirty_equals/_other.py [hunk 1]
```diff
@@ -145,3 +147,85 @@ def is_even(x):
...
+        'query',
+        'fragment',
+    }
+
+    def __init__(
+        self,
+        url_type: Literal[
+            'AnyUrl', 'AnyHttpUrl', 'HttpUrl', 'FileUrl', 'PostgresDsn', 'AmpqpDsn', 'RedisDsn'
+        ] = 'AnyUrl',
+        **expected_attributes: Any,
```
Comment (samuelcolvin) [inline dirty_equals/_other.py:176]: some duplication, but much easier for users if you list each kwarg individually here - so IDEs can provide auto-complete.

File: dirty_equals/_other.py [hunk 2]
```diff
@@ -145,3 +147,85 @@ def is_even(x):
...
+                'IsUrl only checks these Pydantic URL types: AnyUrl, AnyHttpUrl, HttpUrl, FileUrl, '
+                'PostgresDsn, AmpqDsn, RedisDsn'
+            )
+        super().__init__(url_type)
+
+    def equals(self, other: Any) -> bool:
+
+        url_type = self.url_type
+
+        class UserModel(BaseModel):
```
Comment (samuelcolvin) [inline dirty_equals/_other.py:222]: better to use `parse_obj_as`.

File: dirty_equals/_other.py [hunk 3]
```diff
@@ -1,7 +1,9 @@
 import json
-from typing import Any, Callable, TypeVar, overload
+from typing import Any, Callable, Set, TypeVar, overload
 from uuid import UUID
 
+from pydantic import AmqpDsn, AnyHttpUrl, AnyUrl, BaseModel, FileUrl, HttpUrl, PostgresDsn, RedisDsn
```
Comment (samuelcolvin) [inline dirty_equals/_other.py:5]: I don't want to make pydantic a required dependency. Better to import in `IsUrl.__init__` and raise in `ImportError` with instructions on installing `pydantic` if it's not installed.

A bit like [this](https://github.com/pydantic/pydantic/blob/f8575aa3793c565cf006979c60ad62312406d5b5/pydantic/env_settings.py#L328)

**Review by samuelcolvin — COMMENTED** [review]
otherwise good start.

**Comment by osintalex** [comment]
Ready for review!

Sorry I couldn't figure out how to test the optional import of Pydantic throwing an error.

This was the closest I got but it doesn't work since the `mocked_pydantic` object is just imported in `._other.py` rather than called.

```python
def test_import_error_throws():
    import sys
    from unittest.mock import Mock
    mocked_pydantic = Mock(side_effect=ImportError)
    sys.modules['pydantic'] = mocked_pydantic
    with pytest.raises(
            ImportError,
            match='pydantic is not installed, run `pip install dirty-equals[pydantic]`',
    ):
        from dirty_equals import IsUrl
    assert 'https://example.com' == IsUrl
```

**Review by samuelcolvin — APPROVED** [review]
LGTM, thanks so much.

**Pull request merged**